samples/net/tftp_client/README.rst

```diff
@@ -0,0 +1,83 @@
+.. _tftp-client-sample:
+
+TFTP client
+############
+
+Overview
+********
+
+Trivial File Transfer Protocol (TFTP) is a simple lockstep File Transfer Protocol
+based on UDP, and is designed to get a file from or put a file onto a remote host.
+
+This TFTP client sample application for Zephyr implements the TFTP client library
+and establishes a connection (IPv4) to a TFTP server on standard port 69.
+
+The source code for this sample application can be found at:
+:zephyr_file:`samples/net/tftp_client`.
+
+Requirements
+************
+
+- :ref:`networking_with_eth_qemu`, :ref:`networking_with_qemu` or :ref:`networking_with_native_posix`
+- Linux machine
+
+Building and Running
+********************
+
+There are configuration files for various setups in the
+samples/net/tftp_client directory:
+
+- :file:`prj.conf`
+  This is the standard default config.
+
+Build the tftp-client sample application like this:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/tftp_client
+   :board: <board to use>
+   :conf: <config file to use>
+   :goals: build
+   :compact:
+
+The easiest way to setup this sample application is to build and run it
+as native POSIX application or as a QEMU target using the default configuration :file:`prj.conf`.
+This requires a small amount of setup described in :ref:`networking_with_eth_qemu`, :ref:`networking_with_qemu` and :ref:`networking_with_native_posix`.
+
+Build the tftp-client sample application for native_posix like this:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/tftp_client
+   :host-os: unix
+   :board: native_posix
+   :goals: run
+   :compact:
+
+Download and run a TFTP server (like TFTPd), then create file1.bin (with data) and newfile.bin.
+
+To connect to server using hostname, enable dns resolver by changing these two
+configurations in ``prj.conf``::
+
+    CONFIG_DNS_RESOLVER=y
+    CONFIG_NET_CONFIG_PEER_IPV4_ADDR="my-tftp-server.org"
```

Comment:
I think it's a bit counter-intuitive that it's recommended to set ` CONFIG_NET_CONFIG_PEER_IPV4_ADDR` with a hostname. IMO the sample could define it's own Kconfig entry for the server name (which could either by a resolvable hostname or IP address) in a similar way for example `lwm2m_client` does it:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/lwm2m_client/Kconfig#L30